src/kvstore/HBaseEngine.cpp

```diff
@@ -0,0 +1,353 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "base/Base.h"
+#include "kvstore/HBaseEngine.h"
+#include "storage/KeyUtils.h"
+#include "dataman/NebulaCodecImpl.h"
+#include "network/NetworkUtils.h"
+#include "kvstore/KVStore.h"
+
+namespace nebula {
+namespace kvstore {
+
+using nebula::storage::KeyUtils;
+
+const char* kHBaseTableNamePrefix = "Nebula_Graph_Space_";
+const char* kSchemaVersionColumnName = "__sv__";
+const PartitionID kFillPartId = 0;
+const uint8_t kFillMin = 0x00;
+const uint8_t kFillMax = 0xFF;
+const size_t kMaxRowKeyLength = sizeof(PartitionID) + sizeof(VertexID)
+                              + sizeof(EdgeType) + sizeof(VertexID)
+                              + sizeof(EdgeRanking) + sizeof(EdgeVersion);
+
+HBaseEngine::HBaseEngine(GraphSpaceID spaceId,
+                         const HostAddr& hbaseServer,
+                         meta::SchemaManager* schemaMan)
+        : KVEngine(spaceId)
+        , schemaMan_(schemaMan) {
+    CHECK_NOTNULL(schemaMan_);
+    LOG(INFO) << "Connect to the HBase thrift server "
+              << network::NetworkUtils::intToIPv4(hbaseServer.first)
+              << ":" << hbaseServer.second;
+    client_ = std::make_unique<HBaseClient>(hbaseServer);
+    CHECK_NOTNULL(client_.get());
+    tableName_ = kHBaseTableNamePrefix + folly::to<std::string>(spaceId_);
+    partsNum_ = this->allParts().size();
```

Comment:
`partsNum_` is not needed

Reply:
Yes, this is just for overriding interfaces [1] and [2] of KVEngine. If we remove it, that two interfaces may not looks pretty anymore.  And, in fact, the HBaseStore should has **ONE** big partition(partsNum_ = 1). What do you think about it? 

[1] std::vector<PartitionID> allParts() override;   https://github.com/vesoft-inc/nebula/pull/413/files#diff-cc6e1be77ba890ee429b6461fa20a766L70
[2] int32_t totalPartsNum() override;  https://github.com/vesoft-inc/nebula/pull/413/files#diff-cc6e1be77ba890ee429b6461fa20a766L75

Reply:
In the end, I still agree with your suggestion.